src/runtime/entitlements-manager.ts

```diff
@@ -850,25 +753,30 @@ export class EntitlementsManager {
         this.consumeMeter_(entitlement);
       };
 
-      if (!entitlement.subscriptionTokenContents) {
+      const subscriptionTokenContents =
+        entitlement.subscriptionTokenContents as
+          | {
+              metering: {
+                clientType?: MeterClientTypes;
+                clientUserAttribute?: string;
+                showToast: boolean;
+              };
+            }
+          | undefined;
+
+      if (!subscriptionTokenContents) {
         // Ignore decoding errors. Don't show a toast, and return
         // onConsumeCallback directly.
         return onConsumeCallback();
       }
 
-      if (
-        entitlement.subscriptionTokenContents['metering'] &&
-        entitlement.subscriptionTokenContents['metering']['showToast'] === true
-      ) {
+      if (subscriptionTokenContents['metering']['showToast'] === true) {
```

Comment:
This is losing the check that ['metering'] exists, should we use optional chaining here?

Reply:
Nice catch, fixed